### ISSUE TITLE
correct typings missing for `defineNitroPlugin` inside `server/plugins`

### ISSUE THREAD

**Issue by leifmarcus** [nuxt/nuxt#15088]
### Environment

- Operating System: `Darwin`
- Node Version:     `v14.18.0`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `npm@6.14.15`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

I created a simple implementation of a nitro plugin inside the server plugins. The types are all resolved to any: https://stackblitz.com/edit/github-xuiv1i?file=server%2Fplugins%2FNitroPlugin.ts

### Describe the bug

When creating a plugin inside `/server/plugins`, there are no types defined for `defineNitroPlugin`. In comparison if I'm using `defineNuxtPlugin`, it will be typed correctly.

**Comment by kimbaudi** [comment]
@MateiNenciu - I am using nuxt 3.5.3 and `nitroApp` is now typed as `NitroApp`

![image](https://github.com/nuxt/nuxt/assets/1864092/f1b1a12e-4ba9-414d-a6ac-fcada0437466)

However, nitropack seems to be missing typings for `"render:html"`:

![image](https://github.com/nuxt/nuxt/assets/1864092/fdfc8ad8-c8e9-43b5-92d5-cf73847c2a6d)

The typescript errors are:

- `Argument of type '"render:html"' is not assignable to parameter of type 'HookKeys<NitroRuntimeHooks>'.ts(2345)`
- `Property 'head' does not exist on type 'Partial<RenderResponse>'.ts(2339)`

Looking at `node_modules/nitropack/dist/runtime/types.d.ts`, it seems that `NitroRuntimeHooks` interface is missing `"render:html"` property

![image](https://github.com/nuxt/nuxt/assets/1864092/e9b82540-b89f-4dd2-887f-979e517e5091)

**Comment by m3hari** [comment]
Still have this issue on `nuxt 3.12.1` 

```
'"#imports"' has no exported member named 'defineNitroPlugin'. Did you mean 'defineNuxtPlugin'?
```

**Comment by danielroe** [comment]
@m3hari  check you have the correct tsconfig.json in your server/ directory - or open a new issue with a reproduction 🙏

**Comment by itpropro** [comment]
> Still have this issue on `nuxt 3.12.1`
> 
> ```
> '"#imports"' has no exported member named 'defineNitroPlugin'. Did you mean 'defineNuxtPlugin'?
> ```

This is still a problem, the Nitro exports are missing in the #imports in Nuxt. This also applies to `useStorage` for example.

**Comment by danielroe** [comment]
Yes, we may need to use `#nitro-imports` or something similar for server APIs to avoid leakage...

**Comment by arkoannuk** [comment]
Seems to be a relevant issue in ver `3.14`

**Comment by ablaszkiewicz** [comment]
Ran into same issue

**Comment by typed-sigterm** [comment]
Ran into same issue at 4.5.1

**Comment by typed-sigterm** [comment]
Reproduction: https://stackblitz.com/edit/nuxt-demo-d716b6d?file=nuxt.config.ts,server%2Fplugins%2Ft.ts

It requires the long `import { defineNitroPlugin } from 'nitropack/runtime/internal/plugin'` to work. But in my local test, the workaround doesn't work: `RollupError: "defineNitroPlugin" is not exported by "impound:proxy"`

**Comment by typed-sigterm** [comment]
Will it be solved in Nitro v3 stable? Though Nuxt can't work with Nitro v3 now

**Comment by danielroe** [comment]
yes, this is already resolved
